src/group_theory/perm/cycle_type.lean

```diff
@@ -366,13 +367,52 @@ begin
   exact one_lt_two,
 end
 
+lemma exists_prime_order_of_dvd_card {G : Type*} [group G] [fintype G] (p : ℕ) [hp : fact p.prime]
```

Comment:
Did this proof become quite a bit longer?

Reply:
It looks like it's shorter overall because a lot of other lemmas and definitions it previously depended on have been deleted.

Reply:
A little (25 -> 34), but that's because I moved ~50 lines of declarations inside the proof. I could keep the declarations outside if you want, but my impression is that this group action on product-one-vectors is only useful for the proof of Cauchy's theorem. What do you think @jcommelin and @ChrisHughes24 ?

Reply:
I think it's best to keep the vector definition outside of the main theorem just for readability and maintainability even if it's never used. I'm okay with the imports, I hadn't realised the proof had changed.

Reply:
Ok, I'll move them back out.

Reply:
I've moved the vector stuff outside the proof.

Reply:
(Things got a bit longer, because I beefed up the theory by proving `def equiv_vector : vectors_prod_eq_one G n ≃ vector G (n - 1)`